src/sendCommercialMetrics.ts

```diff
@@ -41,6 +42,12 @@ export function sendCommercialMetrics(
 		.map(([name, value]) => ({ name, value: String(value) }))
 		.concat(devProperties);
 
+	if (adBlockerInUse !== undefined)
+		properties.push({
+			name: 'adBlockerInUse',
+			value: adBlockerInUse.toString(),
+		});
+
```

Comment:
If you want to keep the functional style, without mutation, you’d extract this as a `const` and concatenate it like we do fo `devProperties`.

You could enforce this by changing the type on line 40 to be `readonly`:
```ts
const properties: readonly Properties[] = Object.entries(
```

Reply:
Good call! Looks like `CommercialMetrics` also needs to be changed so that `properties` is `readonly`